DEVELOPING.md

```diff
@@ -6,7 +6,64 @@ This is been rewritten to include:
 - Acceptance criteria for code contributions (style, static asserts)
 - How to run the unit tests.
 - How to run the benchmarks.
-- How to run the render tests.
 - How to rebaseline baselines metrics.
 - How to use GL Native as a 3rd party library in your project.
 
+## Render Tests
+
+To check that the output of the rendering is correct, we compare actual rendered PNGs for simple styles with expected PNGs. The content of the tests is stored in the MapLibre GL JS submodule which means that GL JS and Native are in fact quasi pixel-identical in their rendering.
+
+The directory sturcture of the render tests looks like:
+
+```
+maplibre-gl-js/
+  test/
+    integration/
+      render-tests/
+        <name-of-style-spec-feature>/
+          <name-of-feature-value>/
+            expected.png
+            style.json
+```
+
+After the render test run, the folder will also contain an `actual.png` file and a `diff.png` which is the difference between the expected and the actual image. There is a pixel difference threshold value which is used to decide if a render test passed or failed.
+
+### Building the render test runner
+
+Run the following on linux to build the render test runner:
+
+```
+git submodule update --init --recursive --depth 1
+
+sudo apt update
+sudo apt install ccache cmake ninja-build pkg-config xvfb libcurl4-openssl-dev libglfw3-dev libuv1-dev g++-10 libc++-9-dev libc++abi-9-dev libpng-dev libgl1-mesa-dev libgl1-mesa-dri libjpeg-turbo8 libicu66 libjpeg-dev
```

Comment:
Not for now, but we should split basic setup instructions and reder tests.

Also can't we just refer to general build instructions here?

Reply:
Yeah would be better but where are they?

Reply:
This is basically taken from the linux platform readme

Reply:
Let's leave it here for now.